Core/SaveState.cpp

```diff
@@ -567,6 +588,23 @@ namespace SaveState
 		return newestSlot;
 	}
 
+	int GetOldestSlot(const std::string &gameFilename) {
+		int oldestSlot = -1;
+		tm oldestDate = {0};
+		for (int i = 0; i < NUM_SLOTS; i++) {
+			std::string fn = GenerateSaveSlotFilename(gameFilename, i, STATE_EXTENSION);
+			if (File::Exists(fn)) {
+				tm time;
+				bool success = File::GetModifTime(fn, time);
+				if ( (success && !oldestDate) || (success && oldestDate > time) ) {
```

Comment:
nit: Would prefer `(success && (!oldestDate || oldestDate > time))`.

-[Unknown]

Reply:
Yes, that's much better.  Like simplifying like terms in an equation.
Got carried away with overloading the `!` and the `>`.